app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingPage.kt

```diff
@@ -0,0 +1,127 @@
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.button.PrimaryButton
+import org.mozilla.fenix.compose.button.SecondaryButton
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * A page for displaying Onboarding Content.
+ *
+ * @param onboardingPageUiState The page content that's displayed.
+ * @param onPrimaryButtonClick Invoked when the user clicks the primary button.
+ * @param onSecondaryButtonClick Invoked when the user clicks the secondary button.
+ * @param modifier The modifier to be applied to the Composable.
+ */
+@Composable
+fun OnboardingPage(
+    onboardingPageUiState: OnboardingPageUiState,
+    onPrimaryButtonClick: () -> Unit,
+    onSecondaryButtonClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        modifier = modifier.semantics(mergeDescendants = true) {},
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.SpaceEvenly,
+    ) {
+        // Since the vertical arrangement is SpaceEvenly, the text and buttons are wrapped
+        // their own containing Composables so the OuterColumn only has 3 children with
+        // space in between and around them
+        // Item 1
+        Image(
+            painter = painterResource(id = onboardingPageUiState.image),
+            contentDescription = null,
+        )
+
+        // Item 2
+        Column(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalAlignment = Alignment.CenterHorizontally,
+        ) {
+            Text(
+                text = stringResource(id = onboardingPageUiState.title),
+                color = FirefoxTheme.colors.textPrimary,
+                textAlign = TextAlign.Center,
+                style = FirefoxTheme.typography.headline5,
+            )
+
+            Spacer(modifier = Modifier.height(16.dp))
+
+            Text(
+                text = stringResource(id = onboardingPageUiState.description),
+                color = FirefoxTheme.colors.textSecondary,
+                textAlign = TextAlign.Center,
+                style = FirefoxTheme.typography.body2,
+            )
+        }
+
+        // Item 3
+        Column(
+            horizontalAlignment = Alignment.CenterHorizontally,
+        ) {
+            PrimaryButton(
+                text = stringResource(id = onboardingPageUiState.primaryButtonText),
+                onClick = onPrimaryButtonClick,
+            )
+
+            if (onboardingPageUiState.secondaryButtonText != null) {
+                Spacer(modifier = Modifier.height(8.dp))
+                SecondaryButton(
+                    text = stringResource(id = onboardingPageUiState.secondaryButtonText),
+                    onClick = onSecondaryButtonClick,
+                )
+            }
+        }
+    }
+
+    LaunchedEffect(onboardingPageUiState) {
+        // record impression event for onboarding page
```

Comment:
Let's remove the comment here. I think we are repeating the code here with more or less the same exact words.